src/Features/Core/Portable/CodeFixes/Configuration/ConfigurationUpdater.cs

```diff
@@ -294,31 +299,40 @@ private AnalyzerConfigDocument FindOrGenerateEditorConfig()
                 }
             }
 
-            return ImmutableArray<(OptionKey, ICodeStyleOption, IEditorConfigStorageLocation2)>.Empty;
+            return ImmutableArray<(OptionKey, ICodeStyleOption, IEditorConfigStorageLocation2, bool)>.Empty;
         }
 
-        private SourceText GetNewAnalyzerConfigDocumentText(
-            SourceText originalText,
-            Dictionary<string, TextLine> headers)
+        private SourceText GetNewAnalyzerConfigDocumentText(SourceText originalText)
         {
             // Check if an entry to configure the rule severity already exists in the .editorconfig file.
             // If it does, we update the existing entry with the new severity.
-            var configureExistingRuleText = CheckIfRuleExistsAndReplaceInFile(originalText, headers);
-            if (configureExistingRuleText != null)
+            var configureExistingRuleTextAndValidHeader = CheckIfRuleExistsAndReplaceInFile(originalText);
```

Comment:
If you make the change as per suggestion https://github.com/dotnet/roslyn/pull/38992/files#r330735864, then you can write this as `var (newText, lastValidHeaderSpanEnd, lastValidSpecificHeaderSpanEnd) = CheckIfRuleExistsAndReplaceInFile(originalText);`